Sources/NIO/ChannelHandlers.swift

```diff
@@ -17,6 +17,94 @@
 
 import struct Dispatch.DispatchTime
 
+
+/// A `ChannelHandler` that implements a backoff for a `ServerChannel` when accept produces an IOError. `
+/// These errors are often recoverable by reducing the rate at which we call accept.
+public final class AcceptBackoffHandler: ChannelDuplexHandler {
+    public typealias InboundIn = Channel
+    public typealias OutboundIn = Channel
+    
+    private var nextReadDeadlineNS: Int?
+    private let backoffProvider: (IOError) -> TimeAmount?
+    private var scheduledRead: Scheduled<Void>?
+
+    /// Default implementation used as `backoffProvider` which backoff for 1 second.
+    public static func defaultBackoffProvider(error: IOError) -> TimeAmount? {
+        return .seconds(1)
+    }
+
+    /// Create a new instance
+    ///
+    /// - parameters:
+    ///     - backoffProvider: returns a `TimeAmount` to backoff more `reads` (a.k.a accepts).
```

Comment:
Let's reword this to: "returns a `TimeAmount` which will be the amount of time to wait before attempting another `read`"